src/class.cscf_contact.php

```diff
@@ -113,6 +119,13 @@ public function IsValid() {
 			$this->Errors['email'] = esc_html__( 'Please enter a valid email address.', 'clean-and-simple-contact-form-by-meg-nicholas' );
 		}
 
+		//mandatory phone number
+		if ( cscf_PluginSettings::PhoneNumber() && cscf_PluginSettings::PhoneNumberMandatory() ) {
+			if ( strlen( $this->PhoneNumber ) == 0 ) {
```

Comment:
Should we preg_replace first for number spaces dots + and strip the rest first and then check minimum length say 8?

Reply:
I would not enforce any constraints on the phone number, not even a minimum length of 8 characters.

If you contact your local hairdresser or doctor here in Germany, for example, you would not even provide your area code in the contact form. And in small towns its not uncommon to have a five digits telephone number then. Your current implementation would not allow that.

I opt for simply making sure that _anything_ is entered, in case the mandatory option is enabled (i.e. `length > 0`).

Would not strip any characters either. What if people enter their home phone **and** mobile number? We would end up with a messy mix of numbers then. I'd just keep the check very simple and stick to:

`if ( strlen( $this->PhoneNumber ) == 0 ) {`